### PR TITLE
Fix defect magn-6864 List operations on Empty List returns null

### PR DESCRIPTION
**Pull request by ke-yu**
This PR fixes defect mang-6864 that `List.Filter` gets a warning on empty list, and it returns null. 

In ``__Filter()``, it should returns ``{{},{}}``, the first is for filtered in result and the second is for filtered out result.

Language test cases and Dynamo test case added.

@junmendoza please take a look.

### CODE REVIEW

File: test/Engine/ProtoTest/Associative/FunctionObject.cs [hunk 1]
```diff
@@ -403,6 +403,28 @@ def odd(x)
...
+def odd(x)
+{
+    return = x % 2 == 1;
+}
+
+pred = _SingleFunctionObject(odd, 1, { }, { }, true);
+r1 = __Filter({}, pred);
+
+r2 = r1[0];
+r3 = r1[0];
```
Comment (junmendoza) [inline test/Engine/ProtoTest/Associative/FunctionObject.cs]: did you mean r1[1]  ?

File: src/Engine/ProtoCore/ExtendedLibraries/FunctionObject.ds [hunk 2]
```diff
@@ -235,7 +235,7 @@ def __Filter(list: var[]..[], predicate: _FunctionObject)
         _count = Count(list);
         if (_count == 0)
         {
-            return = {};
+            return = {{}, {}};
```
Comment (junmendoza) [inline src/Engine/ProtoCore/ExtendedLibraries/FunctionObject.ds]: Does __Filter always return 2d array for all cases?

Comment (ke-yu) [inline src/Engine/ProtoCore/ExtendedLibraries/FunctionObject.ds]: Yes, unless the input list is null, in that case it should return null.

**Comment by junmendoza** [comment]
Just one minor comment on test, otherwise LGTM

File: test/Engine/ProtoTest/Associative/FunctionObject.cs (hunk 1 above)
Comment (ke-yu) [inline test/Engine/ProtoTest/Associative/FunctionObject.cs]: oh yes, thanks for spotting it!!

**Pull request merged**